### ISSUE TITLE
Object of type 'datetime' is not JSON serializable Failed.

### ISSUE THREAD

**Issue by fazulk** [fishtown-analytics/dbt#2373]
### Describe the bug
Running a basic query that retreives a row(s) from big query. 

These rows contain nested objects which contain a datefield:

`select *  from db`

example row: 
```javascript
"id": "dog", 
"my_object": {
      "domain": "geocities.com",    
      "update_date": "2017-06-02 14:00:00 UTC",     
}
```

### Steps To Reproduce
run `select *  from db` on a table where the row contains a object which contains a date field.

### Expected behavior
the row gets retrieved normally.

### Screenshots and log output
![image](https://user-images.githubusercontent.com/21964900/80621426-20023180-89fc-11ea-94c8-67f24e068cd5.png)

### System information
**Which database are you using dbt with?**
- [ ] postgres
- [ ] redshift
- [x] bigquery
- [ ] snowflake
- [ ] other (specify: ____________)

**The output of `dbt --version`:**
```
DBT Web interface
```

**The operating system you're using:**
Win 10

**The output of `python --version`:**

### Additional context
This same query executes correctly inside Big Query

**Comment by beckjake** [comment]
Thanks for the bug report @fazulk! I'm pretty sure this is a dupe of #2336 (different type, but otherwise the same problem) and is fixed by #2348 - I can't reproduce it locally with the current dev/octavius-catto.

That means this should be resolved in dbt 0.17.0.

**Comment by fazulk** [comment]
perfect, thanks!
